{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d5bf891d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys, os\n",
    "import argparse\n",
    "\n",
    "from Deconvolution.BLADE import Framework\n",
    "import numpy as np\n",
    "from numpy import transpose as t\n",
    "import itertools\n",
    "import pickle\n",
    "from scipy.optimize import nnls\n",
    "from scipy.stats import gaussian_kde\n",
    "from matplotlib.colors import LogNorm\n",
    "from sklearn.svm import SVR\n",
    "from sklearn.svm import NuSVR\n",
    "\n",
    "from sklearn.metrics import mean_squared_error as mse\n",
    "import pandas as pd\n",
    "from tqdm import trange,tqdm\n",
    "# modules for visualization\n",
    "import qgrid\n",
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns\n",
    "import anndata as ad\n",
    "import scanpy as sc\n",
    "import scanorama\n",
    "from sklearn.model_selection import LeaveOneOut,StratifiedKFold,KFold\n",
    "import cycler\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8d5aa6ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scanpy==1.9.1 anndata==0.8.0 umap==0.5.3 numpy==1.21.5 scipy==1.8.0 pandas==1.4.2 scikit-learn==1.0.2 statsmodels==0.13.2 python-igraph==0.9.9 louvain==0.7.1 pynndescent==0.5.7\n"
     ]
    }
   ],
   "source": [
    "sc.settings.verbosity = 3             # verbosity: errors (0), warnings (1), info (2), hints (3)\n",
    "sc.logging.print_header()\n",
    "sc.settings.set_figure_params(dpi=100, facecolor='white')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7331a235",
   "metadata": {},
   "outputs": [],
   "source": [
    "def getloclist(wd,keyword=[\"top\",\"marker\",\"DEG\"]):\n",
    "    loc_list = []\n",
    "    for root, dirs, files in os.walk(wd):\n",
    "        for file in files:\n",
    "            for key in keyword:\n",
    "                if key in file:\n",
    "                    loc = os.path.join(root, file)\n",
    "                    loc_list.append(loc) #get file location\n",
    "                    break\n",
    "    return loc_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1a0eee97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: adata.X seems to be already log-transformed.\n"
     ]
    }
   ],
   "source": [
    "final_adata = sc.read_h5ad(path_adata)\n",
    "final_adata.uns['log1p'][\"base\"] = None\n",
    "final_adata_mean = final_adata.copy()\n",
    "sc.pp.log1p(final_adata_mean)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cccae12b",
   "metadata": {},
   "source": [
    "### Prepare scRNA signature file for CIBERSORTx input\n",
    "Run BayesPrism first to get signature.csv file in place"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "76e4bcfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_CIBERSORTx(path_label,path_signature):\n",
    "    labels = pd.read_csv(path_label+\"cellcategory_simple.csv\",index_col=0)\n",
    "    dict_label = labels.to_dict()['cell_category']\n",
    "    scRNAref =  pd.read_csv(path_signature,index_col=0)\n",
    "    scRNAref.columns = scRNAref.columns.map(dict_label)\n",
    "    scRNAref.to_csv(path_signature.split(\".\")[0]+\"_toCIBERSORTx.tsv\",sep='\\t')\n",
    "    # see preapare_BLADE for test set, signature file from runBP "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1895d8f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# CAUTIONS! this is not part of the script but you need to run it in notebook after the pipeline script finishes\n",
    "list_CIBERSORTx_signature = getloclist(\"/home/cke/PseudoBulk/Results/\",keyword=[\"_signature.tsv\"])\n",
    "for i in list_CIBERSORTx_signature:\n",
    "    prepare_CIBERSORTx(\"/home/cke/Puram/scRNAlabels/\",i)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb50bfe6",
   "metadata": {},
   "source": [
    "## File preparation\n",
    "### Pseudobulk: Cross-validation of dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "936060b8",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def prepare_BLADE(final_adata_mean,final_adata,mode,out):\n",
    "    # Leave-one-out CV\n",
    "#prepare var and mean signature matrix for BLADE\n",
    "    merge_celltype = pd.merge(final_adata_mean.to_df(),final_adata.obs,left_index=True,right_index=True)\n",
    "    if mode == 'real':\n",
    "        counts_Puram_mean = merge_celltype.groupby(['Cell_type']).aggregate(np.mean).fillna(0)[final_adata_mean.to_df().columns]\n",
    "        counts_Puram_std = merge_celltype.groupby(['Cell_type']).aggregate(np.std).fillna(0)[final_adata_mean.to_df().columns]\n",
    "        counts_Puram_mean.T.to_csv(out+mode+\"_mean.tsv\",sep='\\t')\n",
    "        #save signature matrices for TCGA deconv\n",
    "        counts_Puram_std.T.to_csv(out+mode+\"_std.tsv\",sep='\\t')\n",
    "        list_LOT = []\n",
    "\n",
    "    elif mode == 'pseudobulk':\n",
    "        merge_celltype_pseudobulk = merge_celltype.groupby(['batch']).sum()\n",
    "        sample = merge_celltype_pseudobulk.index.tolist()\n",
    "        list_LOT = []\n",
    "        for train_index, test_index in LeaveOneOut().split(sample):\n",
    "            print(\"LOT TRAIN:\", train_index, \"TEST:\", test_index)\n",
    "            train_sample = merge_celltype_pseudobulk.iloc[train_index,].index.tolist()\n",
    "            train = merge_celltype[merge_celltype['batch'].isin(train_sample)]\n",
    "            test_sample = merge_celltype_pseudobulk.iloc[test_index,].index.tolist()\n",
    "            list_LOT.append(test_sample)\n",
    "            print(\"Leaving out: \",test_sample)\n",
    "            test = merge_celltype_pseudobulk[merge_celltype_pseudobulk.index.isin(test_sample)]\n",
    "            counts_Puram_mean = train.groupby(['Cell_type']).aggregate(np.mean).fillna(0)[final_adata_mean.to_df().columns]\n",
    "            counts_Puram_std = train.groupby(['Cell_type']).aggregate(np.std).fillna(0)[final_adata_mean.to_df().columns]\n",
    "            counts_Puram_mean.T.to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_mean.tsv\",sep='\\t')\n",
    "            counts_Puram_std.T.to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_std.tsv\",sep='\\t')\n",
    "            # save the leftout testset for all methods\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_test.tsv\",sep='\\t')\n",
    "            # for CIBERSORTx\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].T.to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_test_transpose.tsv\",sep='\\t')\n",
    "    return list_LOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "4214d2dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_BLADE_kfoldCV(final_adata_mean,final_adata,mode,out,n_splits=5):\n",
    "    #stratified n-fold CV\n",
    "#prepare var and mean signature matrix for BLADE\n",
    "    merge_celltype = pd.merge(final_adata_mean.to_df(),final_adata.obs,left_index=True,right_index=True)\n",
    "    if mode == 'real':\n",
    "        counts_Puram_mean = merge_celltype.groupby(['Cell_type']).aggregate(np.mean).fillna(0)[final_adata_mean.to_df().columns]\n",
    "        counts_Puram_std = merge_celltype.groupby(['Cell_type']).aggregate(np.std).fillna(0)[final_adata_mean.to_df().columns]\n",
    "        counts_Puram_mean.T.to_csv(out+mode+\"_mean.tsv\",sep='\\t')\n",
    "        #save signature matrices for TCGA deconv\n",
    "        counts_Puram_std.T.to_csv(out+mode+\"_std.tsv\",sep='\\t')\n",
    "        list_CV = []\n",
    "\n",
    "    elif mode == 'pseudobulk':\n",
    "        singlecells = merge_celltype.index.tolist()\n",
    "        batch = merge_celltype['batch'].tolist()\n",
    "        list_CV = []\n",
    "        count = 1\n",
    "        for train_index, test_index in StratifiedKFold(n_splits=n_splits).split(singlecells,batch):\n",
    "            print(\"CV TRAIN:\",count , len(train_index), \"TEST:\", len(test_index))\n",
    "            train_sc = merge_celltype.iloc[train_index,].index.tolist()\n",
    "            train = merge_celltype.loc[train_sc,]\n",
    "            test_sc = merge_celltype.iloc[test_index,].index.tolist()\n",
    "            list_CV.append(str(count))\n",
    "            test = merge_celltype.loc[test_sc,].groupby(['batch']).sum()\n",
    "            counts_Puram_mean = train.groupby(['Cell_type']).aggregate(np.mean).fillna(0)[final_adata_mean.to_df().columns]\n",
    "            counts_Puram_std = train.groupby(['Cell_type']).aggregate(np.std).fillna(0)[final_adata_mean.to_df().columns]\n",
    "            counts_Puram_mean.T.to_csv(out+mode+\"_CV\"+str(count)+\"_mean.tsv\",sep='\\t')\n",
    "            counts_Puram_std.T.to_csv(out+mode+\"_CV\"+str(count)+\"_std.tsv\",sep='\\t')\n",
    "            # save the leftout testset for all methods\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_CV\"+str(count)+\"_test.tsv\",sep='\\t')\n",
    "            # for CIBERSORTx\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].T.to_csv(out+mode+\"_CV\"+str(count)+\"_test_transpose.tsv\",sep='\\t')\n",
    "            count+=1\n",
    "    return list_CV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "462d77bc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CV TRAIN: 1 3587 TEST: 897\n",
      "CV TRAIN: 2 3587 TEST: 897\n",
      "CV TRAIN: 3 3587 TEST: 897\n",
      "CV TRAIN: 4 3587 TEST: 897\n",
      "CV TRAIN: 5 3588 TEST: 896\n"
     ]
    }
   ],
   "source": [
    "list_CV = prepare_BLADE_kfoldCV(final_adata_mean,final_adata,'pseudobulk',\"/home/cke/PseudoBulk/InputToWrapper/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "437053ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_Rdeconv(final_adata,mode,out):\n",
    "    merge_sample = pd.merge(final_adata.to_df(),final_adata.obs,left_index=True,right_index=True)\n",
    "    if mode == 'real':\n",
    "        scRNA_input = final_adata.to_df().loc[:,final_adata.to_df().columns]\n",
    "        scRNA_input.to_csv(out+mode+\"_scRNAref.tsv\",sep='\\t')\n",
    "        list_LOT = []\n",
    "        \n",
    "    elif mode == 'pseudobulk':\n",
    "        merge_sample_pseudobulk = merge_sample.groupby(['batch']).sum()\n",
    "        sample = merge_sample_pseudobulk.index.tolist()\n",
    "        list_LOT = []\n",
    "        for train_index, test_index in LeaveOneOut().split(sample):\n",
    "            print(\"LOT TRAIN:\", train_index, \"TEST:\", test_index)\n",
    "            train_sample = merge_sample_pseudobulk.iloc[train_index,].index.tolist()\n",
    "            train = merge_sample[merge_sample['batch'].isin(train_sample)]\n",
    "            test_sample = merge_sample_pseudobulk.iloc[test_index,].index.tolist()\n",
    "            list_LOT.append(test_sample)\n",
    "            print(\"Leaving out: \",test_sample)\n",
    "            test = merge_sample_pseudobulk[merge_sample_pseudobulk.index.isin(test_sample)]\n",
    "            train.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_scRNAtrain.tsv\",sep='\\t')\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_test.tsv\",sep='\\t')\n",
    "            # for CIBERSORTx\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].T.to_csv(out+mode+\"_LOT\"+\"\".join(str(i) for i in test_sample)+\"_test_transpose.tsv\",sep='\\t')\n",
    "    return list_LOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "757d8b12",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_Rdeconv_kfoldCV(final_adata,mode,out,n_splits=5):\n",
    "    merge_sample = pd.merge(final_adata.to_df(),final_adata.obs,left_index=True,right_index=True)\n",
    "    if mode == 'real':\n",
    "        scRNA_input = final_adata.to_df().loc[:,final_adata.to_df().columns]\n",
    "        scRNA_input.to_csv(out+mode+\"_scRNAref.tsv\",sep='\\t')\n",
    "        list_CV = []\n",
    "        \n",
    "    elif mode == 'pseudobulk':\n",
    "        singlecells = merge_celltype.index.tolist()\n",
    "        batch = merge_celltype['batch'].tolist()\n",
    "        list_CV = []\n",
    "        count = 1\n",
    "        for train_index, test_index in StratifiedKFold(n_splits=n_splits).split(singlecells,batch):\n",
    "            print(\"CV TRAIN:\",count , len(train_index), \"TEST:\", len(test_index))\n",
    "            train_sc = merge_celltype.iloc[train_index,].index.tolist()\n",
    "            train = merge_celltype.loc[train_sc,]\n",
    "            test_sc = merge_celltype.iloc[test_index,].index.tolist()\n",
    "            list_CV.append(str(count))\n",
    "            test = merge_celltype.loc[test_sc,].groupby(['batch']).sum()\n",
    "            train.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_CV\"+str(count)+\"_scRNAtrain.tsv\",sep='\\t')\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].to_csv(out+mode+\"_CV\"+str(count)+\"_test.tsv\",sep='\\t')\n",
    "            # for CIBERSORTx\n",
    "            test.iloc[:,:final_adata.to_df().shape[1]].T.to_csv(out+mode+\"_CV\"+str(count)+\"_test_transpose.tsv\",sep='\\t')\n",
    "            count+=1\n",
    "    return list_CV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "0d82b995",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CV TRAIN: 1 3587 TEST: 897\n",
      "CV TRAIN: 2 3587 TEST: 897\n",
      "CV TRAIN: 3 3587 TEST: 897\n",
      "CV TRAIN: 4 3587 TEST: 897\n",
      "CV TRAIN: 5 3588 TEST: 896\n"
     ]
    }
   ],
   "source": [
    "list_CV = prepare_Rdeconv_kfoldCV(final_adata,'pseudobulk',\"/home/cke/PseudoBulk/InputToWrapper/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "8fb4fab9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1', '2', '3', '4', '5']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_CV"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afaa6bc1",
   "metadata": {},
   "source": [
    "### Running each method in cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "943d4ba2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_cmd(mode,out,out_res,dict_FS,folder_marker,\n",
    "            test_sample=False,path_label=\"home/cke/Puram/scRNAlabels/\",\n",
    "            path_bulk=False,name=\"unnamed_job\"):\n",
    "    # \"out\" is folder where input files prepared in last steps\n",
    "    if mode == 'real':\n",
    "        for FS_setup, marker_file in dict_FS.items():\n",
    "            print(mode,\"mode: now with feature selection setup: \\r\",FS_setup)\n",
    "            cmd_MuSiC = \"\".join([\"Rscript /home/cke/runscripts/runMuSiC.r \", \n",
    "                                 out,mode,\"_scRNAref.tsv \", # scRNA signature matrix\n",
    "                                path_bulk,\" \", # real bulk RNAseq matrix\n",
    "                                 path_label,\" \", \n",
    "                              marker_file,\" \",\n",
    "                               name+\"_\"+mode+\"_\"+FS_setup+\" \", # job name FS_setup\n",
    "                              out_res+\"MuSiC/\",\" &\", # output frac folder\n",
    "                                ])\n",
    "            cmd_BP = \"\".join([\"Rscript /home/cke/runscripts/runBP.r \", \n",
    "                                 out,mode,\"_scRNAref.tsv \", # signature matrix\n",
    "                                path_bulk,\" \", # testset pseudobulk matrix\n",
    "                                 path_label,\" \",\n",
    "                              marker_file,\" \",\n",
    "                               name+\"_\"+mode+\"_\"+FS_setup+\" \", # job name FS_setup\n",
    "                              out_res+\"BayesPrism/\",\" \", # output frac folder\n",
    "                              out_res,\" &\" # output CIBERSORTx prelim signature\n",
    "                                ])\n",
    "            print(\"now running in cmd: \\r\",cmd_MuSiC)\n",
    "            print(\"now running in cmd: \\r\",cmd_BP)\n",
    "            os.system(cmd_MuSiC)\n",
    "            os.system(cmd_BP)\n",
    "        # for BLADE, feature selection is done within wrapper\n",
    "        # R is lame, plz use python to develop new tool :)\n",
    "        cmd_BLADE = \"\".join([\"python /home/cke/runscripts/runBLADE.py \", \n",
    "                             out,mode,\"_std.tsv \", # std signature matrix\n",
    "                             out,mode,\"_mean.tsv \", # mean signature matrix\n",
    "                            path_bulk,\" \", # testset pseudobulk matrix\n",
    "                             out_res+\"BLADE/\",\" \", # output folder\n",
    "                             \"--folder_marker \",folder_marker,\" \", # marker folder\n",
    "                             \"--name \",name+\"_\"+mode,\" &\" # job name, background run\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_BLADE)\n",
    "        os.system(cmd_BLADE)\n",
    "    elif mode == \"pseudobulk\":\n",
    "        for FS_setup, marker_file in dict_FS.items():\n",
    "            print(mode,\"mode: now with feature selection setup: \\r\",FS_setup)\n",
    "            cmd_MuSiC = \"\".join([\"Rscript /home/cke/runscripts/runMuSiC.r \", \n",
    "                                 out,mode,\"_CV\",str(test_sample),\"_scRNAtrain.tsv \", # signature matrix\n",
    "                                out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                                 path_label,\" \",\n",
    "                              marker_file,\" \",\n",
    "                               name+\"_\"+mode+\"_\"+FS_setup+\"_\"+str(test_sample)+\"CV \", # job name, FS_setup\n",
    "                              out_res+\"MuSiC/\",\" &\", # output frac folder\n",
    "                                ])\n",
    "            cmd_BP = \"\".join([\"Rscript /home/cke/runscripts/runBP.r \", \n",
    "                                 out,mode,\"_CV\",str(test_sample),\"_scRNAtrain.tsv \", # signature matrix\n",
    "                                out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                                 path_label,\" \",\n",
    "                              marker_file,\" \",\n",
    "                               name+\"_\"+mode+\"_\"+FS_setup+\"_\"+str(test_sample)+\"CV \", # job name, FS_setup\n",
    "                              out_res+\"BayesPrism/\",\" \", # output frac folder\n",
    "                              out_res,\" &\" # output CIBERSORTx prelim signature\n",
    "                                ])\n",
    "            print(\"now running in cmd: \\r\",cmd_MuSiC)\n",
    "            print(\"now running in cmd: \\r\",cmd_BP)\n",
    "            os.system(cmd_MuSiC)\n",
    "            os.system(cmd_BP)\n",
    "        # for BLADE, feature selection is done within wrapper\n",
    "        # R is lame, plz use python to develop new tool :)\n",
    "        cmd_BLADE = \"\".join([\"python /home/cke/runscripts/runBLADE.py \", \n",
    "                             out,mode,\"_CV\",str(test_sample),\"_std.tsv \", # std signature matrix\n",
    "                             out,mode,\"_CV\"+str(test_sample),\"_mean.tsv \", # mean signature matrix\n",
    "                            out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                             out_res+\"BLADE/\",\" \", # output folder\n",
    "                             \"--folder_marker \",folder_marker,\" \", # marker folder\n",
    "                             \"--name \",name+\"_\"+mode+\"_\"+str(test_sample)+\"_CV\",\" &\" # job name, background run\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_BLADE)\n",
    "        os.system(cmd_BLADE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c52738df",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_cmd_noFS(mode,out,out_res,test_sample=False,\n",
    "                 path_label=\"home/cke/Puram/scRNAlabels/\",\n",
    "                 path_bulk=False,name=\"unnamed_noFS_job\"):\n",
    "    # \"out\" is folder where input files prepared in last steps\n",
    "    marker_file = \"noFS\"\n",
    "    if mode == 'real':\n",
    "        print(mode,\"mode: no feature selection\\r\")\n",
    "        cmd_MuSiC = \"\".join([\"Rscript /home/cke/runscripts/runMuSiC.r \", \n",
    "                             out,mode,\"_scRNAref.tsv \", # scRNA signature matrix\n",
    "                            path_bulk,\" \", # real bulk RNAseq matrix\n",
    "                             path_label,\" \", \n",
    "                          marker_file,\" \",\n",
    "                           name+\"_\"+mode+\"_\"+\"noFS\"+\" \", # job name, FS_setup\n",
    "                          out_res+\"MuSiC/\",\" &\", # output frac folder\n",
    "                            ])\n",
    "        cmd_BP = \"\".join([\"Rscript /home/cke/runscripts/runBP.r \", \n",
    "                             out,mode,\"_scRNAref.tsv \", # signature matrix\n",
    "                            path_bulk,\" \", # testset pseudobulk matrix\n",
    "                             path_label,\" \",\n",
    "                          marker_file,\" \",\n",
    "                           name+\"_\"+mode+\"_\"+\"noFS\"+\" \", # job name FS_setup\n",
    "                          out_res+\"BayesPrism/\",\" \", # output frac folder\n",
    "                          out_res,\" &\" # output CIBERSORTx prelim signature\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_MuSiC)\n",
    "        print(\"now running in cmd: \\r\",cmd_BP)\n",
    "        os.system(cmd_MuSiC)\n",
    "        os.system(cmd_BP)\n",
    "        # for BLADE, feature selection is done within wrapper\n",
    "        # R is lame, plz use python to develop new tool :)\n",
    "        cmd_BLADE = \"\".join([\"python /home/cke/runscripts/runBLADE.py \", \n",
    "                             out,mode,\"_std.tsv \", # std signature matrix\n",
    "                             out,mode,\"_mean.tsv \", # mean signature matrix\n",
    "                            path_bulk,\" \", # testset pseudobulk matrix\n",
    "                             out_res+\"BLADE/\",\" \", # output folder\n",
    "                             \"--name \",name+\"_\"+mode,\"_noFS &\" # job name, background run\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_BLADE)\n",
    "        os.system(cmd_BLADE)\n",
    "    elif mode == \"pseudobulk\":\n",
    "        print(mode,\"mode: no feature selection\\r\")\n",
    "        cmd_MuSiC = \"\".join([\"Rscript /home/cke/runscripts/runMuSiC.r \", \n",
    "                             out,mode,\"_CV\",str(test_sample),\"_scRNAtrain.tsv \", # signature matrix\n",
    "                            out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                             path_label,\" \",\n",
    "                          marker_file,\" \",\n",
    "                           name+\"_\"+mode+\"_\"+\"noFS\"+\"_\"+str(test_sample)+\"_CV \", # job name, FS_setup\n",
    "                          out_res+\"MuSiC/\",\" &\", # output frac folder\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_MuSiC)\n",
    "        cmd_BP = \"\".join([\"Rscript /home/cke/runscripts/runBP.r \", \n",
    "                             out,mode,\"_CV\",str(test_sample),\"_scRNAtrain.tsv \", # signature matrix\n",
    "                            out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                             path_label,\" \",\n",
    "                          marker_file,\" \",\n",
    "                           name+\"_\"+mode+\"_\"+\"noFS\"+\"_\"+str(test_sample)+\"_CV \", # job name, FS_setup\n",
    "                          out_res+\"BayesPrism/\",\" \", # output frac folder\n",
    "                          out_res,\" &\" # output CIBERSORTx prelim signature\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_BP)\n",
    "        os.system(cmd_MuSiC)\n",
    "        os.system(cmd_BP)\n",
    "        # for BLADE, feature selection is done within wrapper\n",
    "        # R is lame, plz use python to develop new tool :)\n",
    "        cmd_BLADE = \"\".join([\"python /home/cke/runscripts/runBLADE.py \", \n",
    "                             out,mode,\"_CV\",str(test_sample),\"_std.tsv \", # std signature matrix\n",
    "                             out,mode,\"_CV\"+str(test_sample),\"_mean.tsv \", # mean signature matrix\n",
    "                            out,mode,\"_CV\",str(test_sample),\"_test.tsv \", # testset pseudobulk matrix\n",
    "                             out_res+\"BLADE/\",\" \", # output folder\n",
    "                             \"--name \",name+\"_\"+mode+\"_\"+str(test_sample)+\"_CV_noFS\",\" &\" # job name, background run\n",
    "                            ])\n",
    "        print(\"now running in cmd: \\r\",cmd_BLADE)\n",
    "        os.system(cmd_BLADE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "fd95001d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def main(path_adata,mode,out,out_res,path_label,path_bulk=False,folder_marker=False,\n",
    "         name=\"unnamed_job\",keyword=[\"top\",\"marker\",\"DEG\"]):\n",
    "    # \"out\" is folder where input files prepared in last steps\n",
    "#     print(folder_marker)\n",
    "    final_adata = sc.read_h5ad(path_adata)\n",
    "    final_adata.uns['log1p'][\"base\"] = None\n",
    "    final_adata_mean = final_adata.copy()\n",
    "    sc.pp.log1p(final_adata_mean)\n",
    "    if folder_marker:\n",
    "        print(\"running Feature selection!\\r\")\n",
    "        list_markers = getloclist(folder_marker,keyword)\n",
    "        dict_FS = {}\n",
    "        for marker_file in list_markers:\n",
    "#             marker_genes =  pd.read_csv(marker_file,header=None).iloc[0,:]\n",
    "            dict_FS[os.path.split(marker_file)[1].split(\"_\")[0]] = marker_file\n",
    "            \n",
    "        if mode == 'real':\n",
    "            list_CV = prepare_BLADE_kfoldCV(final_adata_mean,final_adata,mode,out)\n",
    "            list_CV = prepare_Rdeconv_kfoldCV(final_adata,mode,out)\n",
    "            run_cmd(mode,out,out_res,dict_FS,folder_marker,\n",
    "                    path_label=path_label,path_bulk=path_bulk,name=name)\n",
    "            \n",
    "        elif mode == 'pseudobulk':\n",
    "            list_CV = prepare_BLADE_kfoldCV(final_adata_mean,final_adata,mode,out)\n",
    "            list_CV = prepare_Rdeconv_kfoldCV(final_adata,mode,out)\n",
    "            for test_sample in list_CV:\n",
    "                run_cmd(mode,out,out_res,dict_FS,folder_marker,\n",
    "                        test_sample=test_sample,path_label=path_label,name=name)\n",
    "#                 if True: \n",
    "#                     prepare_CIBERSORTx(path_label,path_signature)\n",
    "    else:\n",
    "        print(\"no Feature selection!\\r\")\n",
    "        if mode == 'real':\n",
    "            list_CV = prepare_BLADE_kfoldCV(final_adata_mean,final_adata,mode,out)\n",
    "            list_CV = prepare_Rdeconv_kfoldCV(final_adata,mode,out)\n",
    "            run_cmd_noFS(mode,out,out_res,\n",
    "                    path_label=path_label,path_bulk=path_bulk,name=name)\n",
    "        elif mode == 'pseudobulk':\n",
    "            list_CV = prepare_BLADE_kfoldCV(final_adata_mean,final_adata,mode,out)\n",
    "            list_CV = prepare_Rdeconv_kfoldCV(final_adata,mode,out)\n",
    "            for test_sample in list_CV:\n",
    "                run_cmd_noFS(mode,out,out_res,\n",
    "                        test_sample=test_sample,path_label=path_label,name=name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "e239c12f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_args():\n",
    "    \"\"\"\n",
    "        Parses inputs from the commandline.\n",
    "        :return: inputs as a Namespace object\n",
    "    \"\"\"\n",
    "    parser = argparse.ArgumentParser(description='Generates pipeline')\n",
    "    # Arguments\n",
    "    parser.add_argument('path_adata', help='directory of preprocessed raw scRNA anndata object')\n",
    "    parser.add_argument('mode', help='scheme for data processing', choices=['pseudobulk','real'])\n",
    "    parser.add_argument('out', help='output CV input directory')\n",
    "    parser.add_argument('out_res', help='output of decon methods directory')\n",
    "    parser.add_argument('path_label', help='labels of single-cell type identity directory')\n",
    "    parser.add_argument('--path_bulk', help='bulk rnaseq data directory',default=False)\n",
    "    parser.add_argument('--folder_marker', help='the folder where markers is stored',default=False)\n",
    "    parser.add_argument('--name', help='give this job a name to help remember',defualt='unnamed_job')\n",
    "    parser.add_argument('--keyword', help='keyword in marker file name to identify them',default=[\"top\",\"marker\",\"DEG\"])\n",
    "    return parser.parse_args()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec6a9ab1",
   "metadata": {},
   "outputs": [],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    args = parse_args()\n",
    "    path_adata = args.path_adata\n",
    "    mode = args.mode\n",
    "    out = args.out\n",
    "    out_res = args.out_res\n",
    "    path_label = args.path_label\n",
    "    path_bulk = args.path_bulk\n",
    "    name = args.name\n",
    "    folder_marker = args.folder_marker\n",
    "    keyword = args.keyword\n",
    "    if name == \"unnamed_job\":\n",
    "        print(\"You did not name this job!\\r\")\n",
    "    else:\n",
    "        print(name,\" - pipeline initiated! Welcome, contact author for support: kechanglin1998@hotmail.com\\r\")\n",
    "\n",
    "    if mode == \"real\":\n",
    "        if path_bulk == False:\n",
    "            raise ValueError(\"no bulk RNAseq data input! LOAD UP YOUR AMMO!\\r\")\n",
    "        main(path_adata,mode,out,out_res,path_label,path_bulk=path_bulk,folder_marker=folder_marker,\n",
    "         name=name,keyword=keyword)\n",
    "    elif mode == 'pseudobulk':\n",
    "        main(path_adata,mode,out,out_res,path_label,path_bulk=path_bulk,folder_marker=folder_marker,\n",
    "         name=name,keyword=keyword)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "cca6ea5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# testing code block\n",
    "path_adata = \"/home/cke/Puram/Puram_scanpy.h5ad\"\n",
    "mode = \"real\"\n",
    "out = \"/home/cke/PseudoBulk/InputToWrapper/\"\n",
    "out_res = \"/home/cke/Real/TCGAResults/\"\n",
    "path_label = \"/home/cke/Puram/scRNAlabels/\"\n",
    "path_bulk = \"/home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv\"\n",
    "name = \"test_fullpipeline_1\"\n",
    "folder_marker = \"/home/cke/Puram/markers/\" #only 20 marker file is ready now to save time\n",
    "\n",
    "keyword = [\"top\",\"marker\",\"DEG\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "09cebbe9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# testing code block\n",
    "path_adata = \"/home/cke/Puram/Puram_scanpy.h5ad\"\n",
    "mode = \"pseudobulk\"\n",
    "out = \"/home/cke/PseudoBulk/InputToWrapper/\"\n",
    "out_res = \"/home/cke/PseudoBulk/Results/\"\n",
    "path_label = \"/home/cke/Puram/scRNAlabels/\"\n",
    "path_bulk = False\n",
    "name = \"test_fullpipeline_1\"\n",
    "folder_marker = \"/home/cke/Puram/markers/\" #only 20 marker file is ready now to save time\n",
    "\n",
    "keyword = [\"top\",\"marker\",\"DEG\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "99df82fb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "test_fullpipeline_1  - pipeline initiated! Welcome, contact author for support: kechanglin1998@hotmail.com\n",
      "WARNING: adata.X seems to be already log-transformed.\n",
      "running Feature selection!\n",
      " top20markersw with feature selection setup: \n",
      " Rscript /home/cke/runscripts/runMuSiC.r /home/cke/PseudoBulk/InputToWrapper/real_scRNAref.tsv /home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv /home/cke/Puram/scRNAlabels/ /home/cke/Puram/markers/top20markers_de_cor_symbol.txt test_fullpipeline_1_real_top20markers /home/cke/Real/TCGAResults/MuSiC/ &\n",
      " Rscript /home/cke/runscripts/runBP.r /home/cke/PseudoBulk/InputToWrapper/real_scRNAref.tsv /home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv /home/cke/Puram/scRNAlabels/ /home/cke/Puram/markers/top20markers_de_cor_symbol.txt test_fullpipeline_1_real_top20markers /home/cke/Real/TCGAResults/BayesPrism/ /home/cke/Real/TCGAResults/ &\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: nnls\n",
      "Loading required package: ggplot2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "now running in cmd: \r",
      " python /home/cke/runscripts/runBLADE.py /home/cke/PseudoBulk/InputToWrapper/real_std.tsv /home/cke/PseudoBulk/InputToWrapper/real_mean.tsv /home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv /home/cke/Real/TCGAResults/BLADE/ --folder_marker /home/cke/Puram/markers/ --name test_fullpipeline_1_real &\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: snowfall\n",
      "Loading required package: snow\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Attaching package: ‘reshape’\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    rename\n",
      "\n",
      "corrplot 0.92 loaded\n",
      "\n",
      "Attaching package: ‘tidyr’\n",
      "\n",
      "The following objects are masked from ‘package:reshape’:\n",
      "\n",
      "    expand, smiths\n",
      "\n",
      "Loading required package: lattice\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"running BayesPrism with following args:\"\n",
      "[1] \"/home/cke/PseudoBulk/InputToWrapper/real_scRNAref.tsv\"       \n",
      "[2] \"/home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv\"\n",
      "[3] \"/home/cke/Puram/scRNAlabels/\"                                \n",
      "[4] \"/home/cke/Puram/markers/top20markers_de_cor_symbol.txt\"      \n",
      "[5] \"test_fullpipeline_1_real_top20markers\"                       \n",
      "[6] \"/home/cke/Real/TCGAResults/MuSiC/\"                           \n",
      "[7] \"/home/cke/Puram/scRNAlabels/\"                                \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘gplots::lowess’ by ‘stats::lowess’ when loading ‘TED’ \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"running BayesPrism with following args:\"\n",
      "[1] \"/home/cke/PseudoBulk/InputToWrapper/real_scRNAref.tsv\"       \n",
      "[2] \"/home/cke/TCGA-HNSC.htseq_counts_exp2_symbol_samplexgene.tsv\"\n",
      "[3] \"/home/cke/Puram/scRNAlabels/\"                                \n",
      "[4] \"/home/cke/Puram/markers/top20markers_de_cor_symbol.txt\"      \n",
      "[5] \"test_fullpipeline_1_real_top20markers\"                       \n",
      "[6] \"/home/cke/Real/TCGAResults/BayesPrism/\"                      \n",
      "[7] \"/home/cke/Real/TCGAResults/\"                                 \n",
      "[8] \"/home/cke/Puram/scRNAlabels/\"                                \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Using backend LokyBackend with 10 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "now with feature selection setup:  top20markers\n",
      "Get mean and std exp!\n",
      "Get common genes!  189\n",
      "cell types:  10\n",
      "bulk samples:  546\n",
      "start BLADE!\n",
      "all of 189 genes are used for optimization.\n",
      "All samples are used during the optimization.\n",
      "Initialization with Support vector regression\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:    2.4s\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:    3.8s\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:    5.3s\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:    7.6s\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:   10.1s\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:   12.2s\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:   15.7s\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:   18.6s\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:   21.3s\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:   24.2s\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:   29.3s\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:   32.9s\n",
      "[Parallel(n_jobs=10)]: Done 161 tasks      | elapsed:   36.8s\n",
      "[Parallel(n_jobs=10)]: Done 180 tasks      | elapsed:   40.5s\n",
      "[Parallel(n_jobs=10)]: Done 201 tasks      | elapsed:   45.1s\n",
      "[Parallel(n_jobs=10)]: Done 222 tasks      | elapsed:   50.3s\n",
      "[Parallel(n_jobs=10)]: Done 245 tasks      | elapsed:   55.4s\n",
      "[Parallel(n_jobs=10)]: Done 268 tasks      | elapsed:   59.7s\n",
      "[Parallel(n_jobs=10)]: Done 293 tasks      | elapsed:  1.1min\n",
      "[Parallel(n_jobs=10)]: Done 318 tasks      | elapsed:  1.2min\n",
      "[Parallel(n_jobs=10)]: Done 345 tasks      | elapsed:  1.3min\n",
      "Read 194 items\n",
      "[Parallel(n_jobs=10)]: Done 372 tasks      | elapsed:  1.5min\n",
      "Read 194 items\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Gene symbols detected. Cleaning up genes based on gene symbols. Recommend to use EMSEMBLE IDs for more unique mapping.\"\n",
      "A total of  8  genes from RB chrM chrX chrY  have been excluded \n",
      "A total of  0  lowly expressed genes have been excluded \n",
      "[1] \"removing non-numeric genes...\"\n",
      "[1] \"removing outlier genes...\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Creating Relative Abundance Matrix...\n",
      "[Parallel(n_jobs=10)]: Done 401 tasks      | elapsed:  1.6min\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of outlier genes filtered= 7 \n",
      "[1] \"aligning reference and mixture...\"\n",
      "[1] \"run first sampling\"\n",
      "Start run... This may take a while \n",
      "R Version:  R version 4.1.3 (2022-03-10) \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 10 CPUs.\n",
      "\n",
      "[Parallel(n_jobs=10)]: Done 430 tasks      | elapsed:  1.7min\n",
      "Creating Variance Matrix...\n",
      "[Parallel(n_jobs=10)]: Done 461 tasks      | elapsed:  1.9min\n",
      "Creating Library Size Matrix...\n",
      "Used 189 common genes...\n",
      "Used 10 cell types in deconvolution...\n",
      "TCGA-BB-4224-01A has common genes 188 ...\n",
      "TCGA-H7-7774-01A has common genes 185 ...\n",
      "TCGA-CV-6943-01A has common genes 185 ...\n",
      "TCGA-CN-5374-01A has common genes 187 ...\n",
      "TCGA-CQ-6227-01A has common genes 187 ...\n",
      "TCGA-CV-6959-01A has common genes 186 ...\n",
      "TCGA-F7-A61V-01A has common genes 183 ...\n",
      "TCGA-CV-7413-01A has common genes 185 ...\n",
      "TCGA-CV-7247-01A has common genes 184 ...\n",
      "TCGA-CR-5249-01A has common genes 186 ...\n",
      "TCGA-CQ-5331-01A has common genes 186 ...\n",
      "TCGA-BB-8601-01A has common genes 184 ...\n",
      "TCGA-CV-A45W-01A has common genes 189 ...\n",
      "TCGA-CQ-7063-01A has common genes 183 ...\n",
      "TCGA-CN-5373-01A has common genes 185 ...\n",
      "TCGA-CV-A45T-01A has common genes 186 ...\n",
      "TCGA-HD-A4C1-01A has common genes 186 ...\n",
      "TCGA-CV-7438-11A has common genes 185 ...\n",
      "TCGA-CV-5444-01A has common genes 186 ...\n",
      "TCGA-BA-5556-01A has common genes 185 ...\n",
      "TCGA-P3-A6SX-01A has common genes 184 ...\n",
      "TCGA-CV-7242-11A has common genes 185 ...\n",
      "TCGA-DQ-7592-01A has common genes 187 ...\n",
      "TCGA-CV-6937-01A has common genes 184 ...\n",
      "TCGA-QK-A8ZB-01A has common genes 184 ...\n",
      "TCGA-CV-7432-01A has common genes 183 ...\n",
      "TCGA-CV-7089-01A has common genes 183 ...\n",
      "TCGA-CV-6939-11A has common genes 182 ...\n",
      "TCGA-BA-4075-01A has common genes 185 ...\n",
      "TCGA-T3-A92M-01A has common genes 185 ...\n",
      "TCGA-BA-4078-01A has common genes 187 ...\n",
      "TCGA-CV-7100-01A has common genes 188 ...\n",
      "TCGA-MZ-A7D7-01A has common genes 186 ...\n",
      "TCGA-QK-A652-01A has common genes 185 ...\n",
      "TCGA-CV-6433-01A has common genes 185 ...\n",
      "TCGA-CV-7416-01A has common genes 185 ...\n",
      "TCGA-CV-A6JY-01A has common genes 184 ...\n",
      "TCGA-CR-7399-01A has common genes 182 ...\n",
      "TCGA-CV-7568-01A has common genes 186 ...\n",
      "TCGA-KU-A66S-01A has common genes 187 ...\n",
      "TCGA-QK-A6VC-01A has common genes 189 ...\n",
      "TCGA-CN-6998-01A has common genes 184 ...\n",
      "TCGA-CV-6941-01A has common genes 188 ...\n",
      "TCGA-CN-4727-01A has common genes 183 ...\n",
      "TCGA-BB-A5HY-01A has common genes 186 ...\n",
      "TCGA-CV-7263-01A has common genes 187 ...\n",
      "TCGA-H7-A76A-01A has common genes 188 ...\n",
      "TCGA-CN-4723-01A has common genes 186 ...\n",
      "TCGA-CN-6997-01A has common genes 187 ...\n",
      "TCGA-CN-5364-01A has common genes 183 ...\n",
      "TCGA-CN-A63V-01A has common genes 188 ...\n",
      "TCGA-CQ-5325-01A has common genes 183 ...\n",
      "TCGA-CR-7386-01A has common genes 181 ...\n",
      "TCGA-CX-A4AQ-01A has common genes 185 ...\n",
      "TCGA-QK-A64Z-01A has common genes 185 ...\n",
      "TCGA-BB-4225-01A has common genes 187 ...\n",
      "TCGA-CV-7183-11A has common genes 181 ...\n",
      "TCGA-T2-A6WZ-01A has common genes 185 ...\n",
      "TCGA-HD-8314-01A has common genes 188 ...\n",
      "TCGA-CR-6473-01A has common genes 184 ...\n",
      "TCGA-CN-A642-01A has common genes 183 ...\n",
      "TCGA-BB-A6UO-01A has common genes 184 ...\n",
      "TCGA-DQ-7591-01A has common genes 188 ...\n",
      "TCGA-CR-7398-01A has common genes 187 ...\n",
      "TCGA-HD-8634-01A has common genes 186 ...\n",
      "TCGA-CR-7372-01A has common genes 185 ...\n",
      "TCGA-HD-8224-01A has common genes 185 ...\n",
      "TCGA-BA-4074-01A has common genes 187 ...\n",
      "TCGA-CV-6961-11A has common genes 183 ...\n",
      "TCGA-BB-A5HZ-01A has common genes 185 ...\n",
      "TCGA-CQ-5332-01A has common genes 186 ...\n",
      "TCGA-CR-7380-01A has common genes 184 ...\n",
      "TCGA-CV-6936-01A has common genes 188 ...\n",
      "TCGA-MZ-A6I9-01A has common genes 185 ...\n",
      "TCGA-QK-AA3J-01A has common genes 187 ...\n",
      "TCGA-BA-4077-01B has common genes 185 ...\n",
      "TCGA-CQ-A4C7-01A has common genes 184 ...\n",
      "TCGA-CV-7422-01A has common genes 188 ...\n",
      "TCGA-CQ-A4CB-01A has common genes 186 ...\n",
      "TCGA-F7-7848-01A has common genes 187 ...\n",
      "TCGA-CR-7392-01A has common genes 183 ...\n",
      "TCGA-CV-7178-11A has common genes 184 ...\n",
      "TCGA-QK-A6VB-01A has common genes 188 ...\n",
      "TCGA-CX-7082-01A has common genes 188 ...\n",
      "TCGA-CV-7414-01A has common genes 185 ...\n",
      "TCGA-CN-A63W-01A has common genes 183 ...\n",
      "TCGA-BA-5557-01A has common genes 182 ...\n",
      "TCGA-CV-7435-01A has common genes 186 ...\n",
      "TCGA-UF-A7JT-01A has common genes 189 ...\n",
      "TCGA-QK-A6IF-01A has common genes 186 ...\n",
      "TCGA-QK-A6IJ-01A has common genes 184 ...\n",
      "TCGA-CV-7248-01A has common genes 188 ...\n",
      "TCGA-CV-7101-11A has common genes 185 ...\n",
      "TCGA-H7-A6C4-11A has common genes 183 ...\n",
      "TCGA-CV-6942-01A has common genes 188 ...\n",
      "TCGA-RS-A6TP-01A has common genes 184 ...\n",
      "TCGA-CV-7091-11A has common genes 185 ...\n",
      "TCGA-D6-A6EM-01A has common genes 185 ...\n",
      "TCGA-CV-7101-01A has common genes 187 ...\n",
      "TCGA-CR-7371-01A has common genes 188 ...\n",
      "TCGA-CQ-5329-01A has common genes 185 ...\n",
      "TCGA-H7-8502-01A has common genes 182 ...\n",
      "TCGA-CN-6989-01A has common genes 187 ...\n",
      "TCGA-CN-5360-01A has common genes 188 ...\n",
      "TCGA-CV-7097-11A has common genes 187 ...\n",
      "TCGA-CV-7429-01A has common genes 186 ...\n",
      "TCGA-H7-A6C4-01A has common genes 184 ...\n",
      "TCGA-BA-5153-01A has common genes 185 ...\n",
      "TCGA-BB-8596-01A has common genes 187 ...\n",
      "TCGA-CR-7369-01A has common genes 185 ...\n",
      "TCGA-CQ-7071-01A has common genes 182 ...\n",
      "TCGA-BA-A6DB-01A has common genes 184 ...\n",
      "TCGA-CV-A6K1-01A has common genes 183 ...\n",
      "TCGA-MT-A51X-01A has common genes 183 ...\n",
      "TCGA-4P-AA8J-01A has common genes 186 ...\n",
      "TCGA-CV-5439-01A has common genes 185 ...\n",
      "TCGA-BA-A4IF-01A has common genes 187 ...\n",
      "TCGA-D6-A4Z9-01A has common genes 187 ...\n",
      "TCGA-CV-7437-01A has common genes 184 ...\n",
      "TCGA-CR-6484-01A has common genes 187 ...\n",
      "TCGA-CV-7245-01A has common genes 185 ...\n",
      "TCGA-CQ-5327-01A has common genes 188 ...\n",
      "TCGA-DQ-7588-01A has common genes 181 ...\n",
      "TCGA-CR-7373-01A has common genes 182 ...\n",
      "TCGA-CX-7085-01A has common genes 187 ...\n",
      "TCGA-F7-A50J-01A has common genes 183 ...\n",
      "TCGA-CQ-6228-01A has common genes 184 ...\n",
      "TCGA-CQ-6222-01A has common genes 185 ...\n",
      "TCGA-IQ-7630-01A has common genes 186 ...\n",
      "TCGA-CV-7103-11A has common genes 187 ...\n",
      "TCGA-D6-A6EP-01A has common genes 185 ...\n",
      "TCGA-CV-7250-11A has common genes 186 ...\n",
      "TCGA-D6-6825-01A has common genes 185 ...\n",
      "TCGA-CV-5432-01A has common genes 183 ...\n",
      "TCGA-CN-6013-01A has common genes 186 ...\n",
      "TCGA-CV-6939-01A has common genes 185 ...\n",
      "TCGA-CR-6477-01A has common genes 187 ...\n",
      "TCGA-WA-A7H4-01A has common genes 186 ...\n",
      "TCGA-CN-5358-01A has common genes 187 ...\n",
      "TCGA-CR-7397-01A has common genes 186 ...\n",
      "TCGA-F7-A61W-01A has common genes 183 ...\n",
      "TCGA-BA-5558-01A has common genes 184 ...\n",
      "TCGA-CV-5976-01A has common genes 188 ...\n",
      "TCGA-CV-6954-01A has common genes 184 ...\n",
      "TCGA-CR-7395-01A has common genes 183 ...\n",
      "TCGA-CV-7411-01A has common genes 183 ...\n",
      "TCGA-D6-A6EQ-01A has common genes 183 ...\n",
      "TCGA-CV-7095-01A has common genes 185 ...\n",
      "TCGA-CV-A463-01A has common genes 182 ...\n",
      "TCGA-CN-4722-01A has common genes 184 ...\n",
      "TCGA-QK-A8Z7-01A has common genes 186 ...\n",
      "TCGA-CV-6961-01A has common genes 187 ...\n",
      "TCGA-CN-5365-01A has common genes 186 ...\n",
      "TCGA-CR-5248-01A has common genes 187 ...\n",
      "TCGA-CV-6962-01A has common genes 185 ...\n",
      "TCGA-D6-6517-01A has common genes 184 ...\n",
      "TCGA-CR-6491-01A has common genes 185 ...\n",
      "TCGA-CV-7242-01A has common genes 186 ...\n",
      "TCGA-CV-A45R-01A has common genes 186 ...\n",
      "TCGA-KU-A66T-01A has common genes 188 ...\n",
      "TCGA-BB-4223-01A has common genes 189 ...\n",
      "TCGA-CV-A468-01A has common genes 186 ...\n",
      "TCGA-D6-6516-01A has common genes 186 ...\n",
      "TCGA-CQ-6219-01A has common genes 185 ...\n",
      "TCGA-CR-6467-01A has common genes 188 ...\n",
      "TCGA-CN-A497-01A has common genes 187 ...\n",
      "TCGA-HD-A6I0-11A has common genes 186 ...\n",
      "TCGA-P3-A5QE-01A has common genes 189 ...\n",
      "TCGA-CV-7102-01A has common genes 185 ...\n",
      "TCGA-CV-5443-01A has common genes 187 ...\n",
      "TCGA-CR-7377-01A has common genes 187 ...\n",
      "TCGA-CN-6016-01A has common genes 182 ...\n",
      "TCGA-BA-A6DG-01A has common genes 184 ...\n",
      "TCGA-CQ-A4CD-01A has common genes 183 ...\n",
      "TCGA-F7-A50G-01A has common genes 185 ...\n",
      "TCGA-T2-A6X2-01A has common genes 181 ...\n",
      "TCGA-BA-5559-01A has common genes 189 ...\n",
      "TCGA-CN-6021-01A has common genes 185 ...\n",
      "TCGA-CN-6020-01A has common genes 185 ...\n",
      "TCGA-BA-A6DJ-01A has common genes 182 ...\n",
      "TCGA-BA-A6D8-01A has common genes 185 ...\n",
      "TCGA-CR-7376-01A has common genes 185 ...\n",
      "TCGA-HD-A633-01A has common genes 183 ...\n",
      "TCGA-BA-6871-01A has common genes 189 ...\n",
      "TCGA-CR-6488-01A has common genes 189 ...\n",
      "TCGA-CV-6959-11A has common genes 185 ...\n",
      "TCGA-CR-7364-01A has common genes 182 ...\n",
      "TCGA-CV-7421-01A has common genes 186 ...\n",
      "TCGA-CQ-5323-01A has common genes 185 ...\n",
      "TCGA-UF-A718-01A has common genes 185 ...\n",
      "TCGA-KU-A6H7-01A has common genes 183 ...\n",
      "TCGA-CN-5363-01A has common genes 185 ...\n",
      "TCGA-CN-4730-01A has common genes 182 ...\n",
      "TCGA-CN-5370-01A has common genes 187 ...\n",
      "TCGA-UF-A71D-01A has common genes 183 ...\n",
      "TCGA-UF-A71E-01A has common genes 181 ...\n",
      "TCGA-CV-7416-11A has common genes 186 ...\n",
      "TCGA-CV-7104-01A has common genes 185 ...\n",
      "TCGA-CV-A45P-01A has common genes 188 ...\n",
      "TCGA-CV-7238-01A has common genes 188 ...\n",
      "TCGA-CV-7099-01A has common genes 184 ...\n",
      "TCGA-CV-A6JO-01B has common genes 183 ...\n",
      "TCGA-CV-5971-01A has common genes 187 ...\n",
      "TCGA-D6-A6EN-01A has common genes 180 ...\n",
      "TCGA-CN-4729-01A has common genes 186 ...\n",
      "TCGA-CV-6960-01A has common genes 184 ...\n",
      "TCGA-IQ-A61H-01A has common genes 184 ...\n",
      "TCGA-CV-6956-01A has common genes 187 ...\n",
      "TCGA-MT-A51W-01A has common genes 183 ...\n",
      "TCGA-CQ-6221-01A has common genes 181 ...\n",
      "TCGA-CV-7245-11A has common genes 189 ...\n",
      "TCGA-CV-6953-01A has common genes 182 ...\n",
      "TCGA-CV-5436-01A has common genes 186 ...\n",
      "TCGA-CN-4738-01A has common genes 187 ...\n",
      "TCGA-BA-A6DA-01A has common genes 185 ...\n",
      "TCGA-CR-6474-01A has common genes 186 ...\n",
      "TCGA-CQ-5334-01A has common genes 182 ...\n",
      "TCGA-HD-8635-01A has common genes 185 ...\n",
      "TCGA-CQ-6225-01A has common genes 186 ...\n",
      "TCGA-CV-7250-01A has common genes 187 ...\n",
      "TCGA-F7-A623-01A has common genes 186 ...\n",
      "TCGA-CN-5355-01A has common genes 187 ...\n",
      "TCGA-CV-6934-01A has common genes 189 ...\n",
      "TCGA-CN-5356-01A has common genes 187 ...\n",
      "TCGA-CV-7406-01A has common genes 187 ...\n",
      "TCGA-CV-6943-11A has common genes 184 ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TCGA-CQ-5324-01A has common genes 182 ...\n",
      "TCGA-CQ-5333-01A has common genes 185 ...\n",
      "TCGA-CN-4737-01A has common genes 187 ...\n",
      "TCGA-C9-A480-01A has common genes 188 ...\n",
      "TCGA-D6-6824-01A has common genes 184 ...\n",
      "TCGA-CN-A63T-01A has common genes 187 ...\n",
      "TCGA-CV-7252-01A has common genes 183 ...\n",
      "TCGA-CV-7255-11A has common genes 187 ...\n",
      "TCGA-CV-6003-01A has common genes 184 ...\n",
      "TCGA-F7-A50I-01A has common genes 184 ...\n",
      "TCGA-CV-7407-01A has common genes 184 ...\n",
      "TCGA-CN-5366-01A has common genes 185 ...\n",
      "TCGA-CN-6011-01A has common genes 180 ...\n",
      "TCGA-F7-A61S-01A has common genes 184 ...\n",
      "TCGA-CV-7433-01A has common genes 184 ...\n",
      "TCGA-CR-6480-01A has common genes 186 ...\n",
      "TCGA-CV-7423-11A has common genes 186 ...\n",
      "TCGA-CV-A45O-01A has common genes 181 ...\n",
      "TCGA-CV-6933-01A has common genes 187 ...\n",
      "TCGA-CV-A45V-01A has common genes 184 ...\n",
      "TCGA-CN-A49C-01A has common genes 186 ...\n",
      "TCGA-DQ-5630-01A has common genes 188 ...\n",
      "TCGA-UP-A6WW-01A has common genes 188 ...\n",
      "TCGA-CV-7406-11A has common genes 188 ...\n",
      "TCGA-P3-A6T5-01A has common genes 184 ...\n",
      "TCGA-CV-7261-11A has common genes 182 ...\n",
      "TCGA-IQ-A61J-01A has common genes 184 ...\n",
      "TCGA-CR-7379-01A has common genes 185 ...\n",
      "TCGA-CV-A45Z-01A has common genes 187 ...\n",
      "TCGA-CN-4742-01A has common genes 183 ...\n",
      "TCGA-CV-A461-01A has common genes 189 ...\n",
      "TCGA-CR-7388-01A has common genes 186 ...\n",
      "TCGA-BA-A6DE-01A has common genes 186 ...\n",
      "TCGA-CV-6948-01A has common genes 188 ...\n",
      "TCGA-BB-4227-01A has common genes 185 ...\n",
      "TCGA-P3-A6T4-01A has common genes 184 ...\n",
      "TCGA-QK-A6V9-01A has common genes 185 ...\n",
      "TCGA-CR-7389-01A has common genes 184 ...\n",
      "TCGA-CV-6935-01A has common genes 187 ...\n",
      "TCGA-CN-6017-01A has common genes 184 ...\n",
      "TCGA-CV-A6JU-01A has common genes 187 ...\n",
      "TCGA-CR-7401-01A has common genes 185 ...\n",
      "TCGA-CV-6933-11A has common genes 186 ...\n",
      "TCGA-CN-4739-01A has common genes 189 ...\n",
      "TCGA-CV-A45Y-01A has common genes 185 ...\n",
      "TCGA-CR-6493-01A has common genes 181 ...\n",
      "TCGA-BA-4076-01A has common genes 182 ...\n",
      "TCGA-CV-6436-01A has common genes 181 ...\n",
      "TCGA-DQ-5625-01A has common genes 185 ...\n",
      "TCGA-HL-7533-01A has common genes 185 ...\n",
      "TCGA-CV-A6JM-01A has common genes 177 ...\n",
      "TCGA-QK-AA3K-01A has common genes 182 ...\n",
      "TCGA-HD-7831-01A has common genes 183 ...\n",
      "TCGA-UF-A71B-01A has common genes 186 ...\n",
      "TCGA-CV-A45U-01A has common genes 186 ...\n",
      "TCGA-CV-7440-11A has common genes 186 ...\n",
      "TCGA-CV-6955-11A has common genes 187 ...\n",
      "TCGA-CV-7253-01A has common genes 186 ...\n",
      "TCGA-P3-A6T8-01A has common genes 186 ...\n",
      "TCGA-CR-5250-01A has common genes 188 ...\n",
      "TCGA-BA-5152-01A has common genes 186 ...\n",
      "TCGA-CR-5247-01A has common genes 184 ...\n",
      "TCGA-MT-A7BN-01A has common genes 189 ...\n",
      "TCGA-CN-A499-01A has common genes 184 ...\n",
      "TCGA-CN-A498-01A has common genes 187 ...\n",
      "TCGA-CV-7103-01A has common genes 187 ...\n",
      "TCGA-T2-A6X0-01A has common genes 188 ...\n",
      "TCGA-CV-A45Q-01A has common genes 172 ...\n",
      "TCGA-DQ-5631-01A has common genes 183 ...\n",
      "TCGA-CV-7410-01A has common genes 187 ...\n",
      "TCGA-P3-A6T2-01A has common genes 185 ...\n",
      "TCGA-CV-6938-01A has common genes 178 ...\n",
      "TCGA-CQ-A4C6-01A has common genes 186 ...\n",
      "TCGA-CV-7425-11A has common genes 183 ...\n",
      "TCGA-BA-6873-01A has common genes 185 ...\n",
      "TCGA-CN-6022-01A has common genes 185 ...\n",
      "TCGA-CV-7235-11A has common genes 184 ...\n",
      "TCGA-CQ-6229-01A has common genes 187 ...\n",
      "TCGA-CV-6441-01A has common genes 181 ...\n",
      "TCGA-CV-7434-01A has common genes 182 ...\n",
      "TCGA-CV-6952-01A has common genes 187 ...\n",
      "TCGA-CQ-5326-01A has common genes 185 ...\n",
      "TCGA-CN-A6V6-01A has common genes 186 ...\n",
      "TCGA-CV-6955-01A has common genes 181 ...\n",
      "TCGA-CN-6019-01A has common genes 188 ...\n",
      "TCGA-CV-7437-11A has common genes 187 ...\n",
      "TCGA-CR-5243-01A has common genes 188 ...\n",
      "TCGA-BA-6868-01B has common genes 187 ...\n",
      "TCGA-CQ-5330-01A has common genes 184 ...\n",
      "TCGA-P3-A6T7-01A has common genes 185 ...\n",
      "TCGA-D6-6827-01A has common genes 187 ...\n",
      "TCGA-CN-6992-01A has common genes 189 ...\n",
      "TCGA-CR-7402-01A has common genes 186 ...\n",
      "TCGA-CR-6470-01A has common genes 188 ...\n",
      "TCGA-CN-6023-01A has common genes 187 ...\n",
      "TCGA-DQ-5624-01A has common genes 186 ...\n",
      "TCGA-HD-A6HZ-11A has common genes 186 ...\n",
      "TCGA-CQ-A4CE-01A has common genes 186 ...\n",
      "TCGA-UF-A7JV-01A has common genes 187 ...\n",
      "TCGA-CV-7427-01A has common genes 185 ...\n",
      "TCGA-CV-5973-01A has common genes 182 ...\n",
      "TCGA-DQ-5629-01A has common genes 183 ...\n",
      "TCGA-BA-A4IH-01A has common genes 187 ...\n",
      "TCGA-CV-A6JT-01A has common genes 185 ...\n",
      "TCGA-CV-7177-01A has common genes 183 ...\n",
      "TCGA-BA-6870-01A has common genes 185 ...\n",
      "TCGA-CV-7446-01A has common genes 186 ...\n",
      "TCGA-IQ-A61G-01A has common genes 182 ...\n",
      "TCGA-D6-A6EK-01A has common genes 183 ...\n",
      "TCGA-IQ-A61I-01A has common genes 186 ...\n",
      "TCGA-CN-A6UY-01A has common genes 186 ...\n",
      "TCGA-CV-A465-01A has common genes 185 ...\n",
      "TCGA-CR-7383-01A has common genes 185 ...\n",
      "TCGA-P3-A6T0-01A has common genes 184 ...\n",
      "TCGA-D6-A4ZB-01A has common genes 185 ...\n",
      "TCGA-BA-A4II-01A has common genes 184 ...\n",
      "TCGA-CN-5359-01A has common genes 188 ...\n",
      "TCGA-CV-7438-01A has common genes 185 ...\n",
      "TCGA-CV-7097-01A has common genes 184 ...\n",
      "TCGA-QK-A8Z9-01B has common genes 185 ...\n",
      "TCGA-UF-A7JO-01A has common genes 187 ...\n",
      "TCGA-CV-7180-01A has common genes 184 ...\n",
      "TCGA-D6-A6EO-01A has common genes 184 ...\n",
      "TCGA-HD-A6I0-01A has common genes 184 ...\n",
      "TCGA-CV-6950-01A has common genes 186 ...\n",
      "TCGA-UF-A7JK-01A has common genes 185 ...\n",
      "TCGA-UF-A7JF-01A has common genes 188 ...\n",
      "TCGA-CV-7428-01A has common genes 187 ...\n",
      "TCGA-D6-A6ES-01A has common genes 175 ...\n",
      "TCGA-CQ-A4CH-01A has common genes 185 ...\n",
      "TCGA-CV-5966-01A has common genes 185 ...\n",
      "TCGA-CR-7394-01A has common genes 184 ...\n",
      "TCGA-CV-7425-01A has common genes 184 ...\n",
      "TCGA-CV-7254-01A has common genes 186 ...\n",
      "TCGA-CR-6478-01A has common genes 184 ...\n",
      "TCGA-TN-A7HL-01A has common genes 188 ...\n",
      "TCGA-CV-7432-11A has common genes 187 ...\n",
      "TCGA-CQ-6218-01A has common genes 184 ...\n",
      "TCGA-HD-7229-01A has common genes 183 ...\n",
      "TCGA-MT-A67D-01A has common genes 188 ...\n",
      "TCGA-IQ-7631-01A has common genes 186 ...\n",
      "TCGA-CQ-7072-01A has common genes 184 ...\n",
      "TCGA-BB-A6UM-01A has common genes 185 ...\n",
      "TCGA-MT-A67F-01A has common genes 183 ...\n",
      "TCGA-CN-6995-01A has common genes 181 ...\n",
      "TCGA-CN-A6V7-01A has common genes 187 ...\n",
      "TCGA-BA-A8YP-01A has common genes 188 ...\n",
      "TCGA-CN-A49B-01A has common genes 185 ...\n",
      "TCGA-CV-7183-01A has common genes 180 ...\n",
      "TCGA-UF-A7JJ-01A has common genes 187 ...\n",
      "TCGA-CR-7385-01A has common genes 187 ...\n",
      "TCGA-P3-A5QF-01A has common genes 184 ...\n",
      "TCGA-H7-A6C5-11A has common genes 185 ...\n",
      "TCGA-H7-8501-01A has common genes 185 ...\n",
      "TCGA-IQ-7632-01A has common genes 180 ...\n",
      "TCGA-CV-A6JZ-01A has common genes 186 ...\n",
      "TCGA-HD-7832-01A has common genes 184 ...\n",
      "TCGA-CV-A6JE-01A has common genes 183 ...\n",
      "TCGA-CN-4725-01A has common genes 186 ...\n",
      "TCGA-BA-5151-01A has common genes 187 ...\n",
      "TCGA-CN-4726-01A has common genes 182 ...\n",
      "TCGA-F7-A624-01A has common genes 183 ...\n",
      "TCGA-BA-A6DL-01A has common genes 186 ...\n",
      "TCGA-BA-6869-01A has common genes 186 ...\n",
      "TCGA-QK-A6IH-01A has common genes 178 ...\n",
      "TCGA-HD-7754-01A has common genes 186 ...\n",
      "TCGA-CV-A6JN-01A has common genes 185 ...\n",
      "TCGA-TN-A7HI-01A has common genes 187 ...\n",
      "TCGA-CR-7365-01A has common genes 184 ...\n",
      "TCGA-CV-5440-01A has common genes 187 ...\n",
      "TCGA-CV-6956-11A has common genes 185 ...\n",
      "TCGA-MZ-A5BI-01A has common genes 184 ...\n",
      "TCGA-BB-4228-01A has common genes 185 ...\n",
      "TCGA-QK-A6II-01A has common genes 180 ...\n",
      "TCGA-CV-7252-11A has common genes 187 ...\n",
      "TCGA-CV-5970-01A has common genes 186 ...\n",
      "TCGA-CQ-7065-01A has common genes 184 ...\n",
      "TCGA-QK-A8Z8-01A has common genes 187 ...\n",
      "TCGA-CR-7382-01A has common genes 186 ...\n",
      "TCGA-CR-7368-01A has common genes 187 ...\n",
      "TCGA-CV-6934-11A has common genes 188 ...\n",
      "TCGA-CQ-A4C9-01A has common genes 187 ...\n",
      "TCGA-CQ-6224-01A has common genes 184 ...\n",
      "TCGA-CN-5369-01A has common genes 187 ...\n",
      "TCGA-CR-7374-01A has common genes 189 ...\n",
      "TCGA-CV-5978-01A has common genes 186 ...\n",
      "TCGA-QK-A8ZA-01A has common genes 188 ...\n",
      "TCGA-D6-6823-01A has common genes 185 ...\n",
      "TCGA-CN-6012-01A has common genes 186 ...\n",
      "TCGA-CV-6945-01A has common genes 186 ...\n",
      "TCGA-CR-7393-01A has common genes 183 ...\n",
      "TCGA-CN-6994-01A has common genes 185 ...\n",
      "TCGA-MT-A67A-01A has common genes 185 ...\n",
      "TCGA-CN-A63U-01A has common genes 186 ...\n",
      "TCGA-CV-7418-01A has common genes 189 ...\n",
      "TCGA-IQ-A6SG-01A has common genes 185 ...\n",
      "TCGA-CV-5977-01A has common genes 181 ...\n",
      "TCGA-BA-6872-01A has common genes 185 ...\n",
      "TCGA-CR-7391-01A has common genes 187 ...\n",
      "TCGA-CV-7415-01A has common genes 183 ...\n",
      "TCGA-F7-8298-01A has common genes 181 ...\n",
      "TCGA-BA-A6DI-01A has common genes 187 ...\n",
      "TCGA-CQ-A4CI-01A has common genes 187 ...\n",
      "TCGA-BB-A5HU-01A has common genes 184 ...\n",
      "TCGA-HD-A634-01A has common genes 188 ...\n",
      "TCGA-CV-5434-01A has common genes 186 ...\n",
      "TCGA-CV-7434-11A has common genes 184 ...\n",
      "TCGA-D6-8568-01A has common genes 185 ...\n",
      "TCGA-CV-A6K0-01B has common genes 186 ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TCGA-HD-8635-11A has common genes 182 ...\n",
      "TCGA-P3-A5Q6-01A has common genes 185 ...\n",
      "TCGA-F7-A620-01A has common genes 185 ...\n",
      "TCGA-CN-A49A-01A has common genes 186 ...\n",
      "TCGA-CV-5430-01A has common genes 186 ...\n",
      "TCGA-F7-8489-01A has common genes 183 ...\n",
      "TCGA-KU-A6H8-01A has common genes 186 ...\n",
      "TCGA-CR-7370-01A has common genes 185 ...\n",
      "TCGA-CN-4735-01A has common genes 183 ...\n",
      "TCGA-CN-A6V3-01A has common genes 188 ...\n",
      "TCGA-CN-5367-01A has common genes 187 ...\n",
      "TCGA-CV-5431-01A has common genes 185 ...\n",
      "TCGA-F7-A622-01A has common genes 184 ...\n",
      "TCGA-UF-A7JA-01A has common genes 185 ...\n",
      "TCGA-CV-A6JD-01A has common genes 185 ...\n",
      "TCGA-CR-6481-01A has common genes 187 ...\n",
      "TCGA-BA-A6DD-01A has common genes 183 ...\n",
      "TCGA-WA-A7GZ-01A has common genes 187 ...\n",
      "TCGA-TN-A7HJ-01A has common genes 188 ...\n",
      "TCGA-UF-A7JC-01A has common genes 183 ...\n",
      "TCGA-P3-A5Q5-01A has common genes 186 ...\n",
      "TCGA-UF-A71A-06A has common genes 187 ...\n",
      "TCGA-D6-6515-01A has common genes 185 ...\n",
      "TCGA-BA-A4IG-01A has common genes 187 ...\n",
      "TCGA-CV-7423-01A has common genes 183 ...\n",
      "TCGA-D6-8569-01A has common genes 185 ...\n",
      "TCGA-CN-6988-01A has common genes 187 ...\n",
      "TCGA-CV-7424-11A has common genes 187 ...\n",
      "TCGA-CN-4740-01A has common genes 188 ...\n",
      "TCGA-CR-6487-01A has common genes 187 ...\n",
      "TCGA-D6-6826-01A has common genes 187 ...\n",
      "TCGA-CN-6018-01A has common genes 185 ...\n",
      "TCGA-CR-7367-01A has common genes 186 ...\n",
      "TCGA-P3-A6T6-01A has common genes 184 ...\n",
      "TCGA-CV-5435-01A has common genes 187 ...\n",
      "TCGA-BA-7269-01A has common genes 186 ...\n",
      "TCGA-UF-A7JD-01A has common genes 187 ...\n",
      "TCGA-CR-6492-01A has common genes 189 ...\n",
      "TCGA-CR-6482-01A has common genes 188 ...\n",
      "TCGA-CV-7440-01A has common genes 184 ...\n",
      "TCGA-CV-A6K2-01A has common genes 182 ...\n",
      "TCGA-CV-5441-01A has common genes 188 ...\n",
      "TCGA-CV-7178-01A has common genes 186 ...\n",
      "TCGA-BA-5555-01A has common genes 184 ...\n",
      "TCGA-UF-A7J9-01A has common genes 188 ...\n",
      "TCGA-CV-7091-01A has common genes 186 ...\n",
      "TCGA-UF-A7JS-01A has common genes 185 ...\n",
      "TCGA-CV-5442-01A has common genes 181 ...\n",
      "TCGA-HD-7753-01A has common genes 186 ...\n",
      "TCGA-CV-6940-01A has common genes 185 ...\n",
      "TCGA-QK-A6IG-01A has common genes 185 ...\n",
      "TCGA-CN-6996-01A has common genes 184 ...\n",
      "TCGA-CN-5361-01A has common genes 186 ...\n",
      "TCGA-IQ-A61E-01A has common genes 183 ...\n",
      "TCGA-RS-A6TO-01A has common genes 182 ...\n",
      "TCGA-CN-A641-01A has common genes 186 ...\n",
      "TCGA-CN-4741-01A has common genes 187 ...\n",
      "TCGA-CV-6936-11A has common genes 186 ...\n",
      "TCGA-CN-4734-01A has common genes 185 ...\n",
      "TCGA-CN-4731-01A has common genes 185 ...\n",
      "TCGA-CQ-A4CG-01A has common genes 186 ...\n",
      "TCGA-CX-7219-01A has common genes 182 ...\n",
      "TCGA-IQ-A61O-01A has common genes 184 ...\n",
      "TCGA-CV-6938-11A has common genes 185 ...\n",
      "TCGA-T2-A6WX-01A has common genes 186 ...\n",
      "TCGA-CQ-7068-01A has common genes 186 ...\n",
      "TCGA-CV-7238-11A has common genes 188 ...\n",
      "TCGA-P3-A6T3-01A has common genes 185 ...\n",
      "TCGA-UF-A7JH-01A has common genes 187 ...\n",
      "TCGA-P3-A5QA-01A has common genes 188 ...\n",
      "TCGA-CV-7424-01A has common genes 187 ...\n",
      "TCGA-CV-7177-11A has common genes 183 ...\n",
      "TCGA-CQ-6220-01A has common genes 185 ...\n",
      "TCGA-UF-A71A-01A has common genes 184 ...\n",
      "TCGA-CR-6472-01A has common genes 186 ...\n",
      "TCGA-T3-A92N-01A has common genes 187 ...\n",
      "TCGA-CR-6471-01A has common genes 185 ...\n",
      "TCGA-CN-6024-01A has common genes 187 ...\n",
      "TCGA-KU-A6H7-06A has common genes 186 ...\n",
      "TCGA-BB-4217-01A has common genes 188 ...\n",
      "TCGA-CQ-A4CA-01A has common genes 185 ...\n",
      "TCGA-CV-7430-01A has common genes 182 ...\n",
      "TCGA-D6-A74Q-01A has common genes 187 ...\n",
      "TCGA-CV-7255-01A has common genes 184 ...\n",
      "TCGA-CQ-6223-01A has common genes 184 ...\n",
      "TCGA-CV-A460-01A has common genes 185 ...\n",
      "TCGA-CV-7261-01A has common genes 187 ...\n",
      "TCGA-CN-4736-01A has common genes 185 ...\n",
      "TCGA-CR-7404-01A has common genes 187 ...\n",
      "TCGA-UF-A719-01A has common genes 188 ...\n",
      "TCGA-CV-7236-01A has common genes 184 ...\n",
      "TCGA-CV-6951-01A has common genes 186 ...\n",
      "TCGA-CN-4728-01A has common genes 188 ...\n",
      "TCGA-CV-7090-01A has common genes 182 ...\n",
      "TCGA-CV-5979-01A has common genes 184 ...\n",
      "TCGA-CQ-7069-01A has common genes 183 ...\n",
      "TCGA-CN-4733-01A has common genes 184 ...\n",
      "TCGA-CR-7390-01A has common genes 185 ...\n",
      "TCGA-CV-A45X-01A has common genes 184 ...\n",
      "TCGA-IQ-A6SH-01A has common genes 183 ...\n",
      "TCGA-CV-6962-11A has common genes 183 ...\n",
      "TCGA-CV-6960-11A has common genes 184 ...\n",
      "TCGA-CV-A464-01A has common genes 186 ...\n",
      "TCGA-C9-A47Z-01A has common genes 184 ...\n",
      "TCGA-CN-6010-01A has common genes 188 ...\n",
      "TCGA-WA-A7GZ-11A has common genes 186 ...\n",
      "TCGA-CV-7235-01A has common genes 186 ...\n",
      "TCGA-CX-7086-01A has common genes 187 ...\n",
      "TCGA-CV-6935-11A has common genes 186 ...\n",
      "TCGA-P3-A6SW-01A has common genes 185 ...\n",
      "TCGA-HD-A6HZ-01A has common genes 186 ...\n",
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file '/home/cke/Real/TCGAResults/MuSiC/celltypefrac_MuSiC_test_fullpipeline_1_real_top20markers.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done 492 tasks      | elapsed:  2.0min\n",
      "[Parallel(n_jobs=10)]: Done 525 tasks      | elapsed:  2.2min\n",
      "[Parallel(n_jobs=10)]: Done 546 out of 546 | elapsed:  2.3min finished\n",
      "[Parallel(n_jobs=10)]: Using backend LokyBackend with 10 concurrent workers.\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:589: NumbaPerformanceWarning: \u001b[1m\u001b[1m\u001b[1m\u001b[1mnp.dot() is faster on contiguous arrays, called on (array(float64, 1d, C), array(float64, 2d, A))\u001b[0m\u001b[0m\u001b[0m\u001b[0m\n",
      "  g_Exp = g_Exp_Beta(Nu, Omega, Beta, B0, Ngene, Ncell, Nsample)\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "\n",
      "Stopping cluster\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"merge subtypes\"\n",
      "        Fibroblast tumor B cell myocyte other Macrophage Endothelial T cell\n",
      "Min.         0.000 0.000  0.000   0.000 0.000      0.000       0.000  0.000\n",
      "1st Qu.      0.034 0.620  0.000   0.000 0.000      0.004       0.008  0.001\n",
      "Median       0.101 0.788  0.002   0.000 0.000      0.011       0.015  0.004\n",
      "Mean         0.148 0.730  0.009   0.029 0.031      0.019       0.021  0.009\n",
      "3rd Qu.      0.204 0.887  0.009   0.011 0.016      0.027       0.027  0.009\n",
      "Max.         0.956 0.999  0.207   0.826 0.993      0.237       0.218  0.119\n",
      "        Dendritic  Mast\n",
      "Min.        0.000 0.000\n",
      "1st Qu.     0.000 0.000\n",
      "Median      0.001 0.000\n",
      "Mean        0.004 0.000\n",
      "3rd Qu.     0.003 0.000\n",
      "Max.        0.089 0.004\n",
      "[1] \"pooling information across samples\"\n",
      "Start optimization... This may take a while \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 10 CPUs.\n",
      "\n",
      "\n",
      "Stopping cluster\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"run final sampling\"\n",
      "Start run... This may take a while \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 10 CPUs.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# testing code block\n",
    "\n",
    "if name == \"unnamed_job\":\n",
    "    print(\"You did not name this job!\\r\")\n",
    "else:\n",
    "    print(name,\" - pipeline initiated! Welcome, contact author for support: kechanglin1998@hotmail.com\\r\")\n",
    "\n",
    "if mode == \"real\":\n",
    "    if path_bulk == False:\n",
    "        raise ValueError(\"no bulk RNAseq data input! LOAD UP YOUR AMMO!\\r\")\n",
    "    main(path_adata,mode,out,out_res,path_label,path_bulk=path_bulk,folder_marker=folder_marker,\n",
    "     name=name,keyword=keyword)\n",
    "elif mode == 'pseudobulk':\n",
    "    main(path_adata,mode,out,out_res,path_label,path_bulk=path_bulk,folder_marker=folder_marker,\n",
    "     name=name,keyword=keyword)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "17610b78",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  4.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  4.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.0min\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.1min\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.1min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.0min\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.1min\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.2min\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.3min\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.2min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.3min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.3min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.2min\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.3min\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.3min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.4min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.4min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.4min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.4min\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.3min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.4min\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.4min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done   5 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.6min\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  5.5min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  5.6min\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  5.5min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.6min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.6min\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.6min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.7min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  5.6min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.7min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  12 tasks      | elapsed:  5.7min\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  5.7min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  5.7min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  5.8min\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  5.7min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  5.7min\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done  21 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.7min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.8min\n",
      "[Parallel(n_jobs=10)]: Done  30 tasks      | elapsed:  5.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  5.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  5.9min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.9min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  5.9min\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  41 tasks      | elapsed:  5.8min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/BayesPrism/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  5.9min\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.0min\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  6.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.0min\n",
      "[Parallel(n_jobs=10)]: Done  52 tasks      | elapsed:  5.9min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  6.0min\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  6.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  6.1min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.1min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.0min\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.0min remaining:    4.5s\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.0min finished\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.1min remaining:    4.6s\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.1min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  65 tasks      | elapsed:  6.0min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  6.1min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.1min finished\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  78 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.2min\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:809: RuntimeWarning: divide by zero encountered in log\n",
      "  return -self.Nsample*np.sum(np.log(Omega))\n",
      "/home/cke/miniconda3/envs/BLADE/lib/python3.10/site-packages/Deconvolution/BLADE.py:863: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return PX+PY+PF-QX-QF\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done  93 tasks      | elapsed:  6.1min\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.3min\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 108 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.3min remaining:    4.7s\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.3min remaining:    4.7s\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.3min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.3min\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.2min\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.3min finished\n",
      "[Parallel(n_jobs=10)]: Done 125 tasks      | elapsed:  6.3min\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.4min\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.3min remaining:    4.8s\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.4min finished\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.3min\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.4min\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.4min remaining:    4.8s\n",
      "[Parallel(n_jobs=10)]: Done 142 tasks      | elapsed:  6.4min\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.4min finished\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.3min remaining:    4.8s\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.4min remaining:    4.8s\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.4min finished\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.3min finished\n",
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  6.4min finished\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.416964999834696\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_28_LOT_BLADEout_top20markers.pickle\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.514795935153961\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_17_LOT_BLADEout_top20markers.pickle\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, \"rt\") : cannot open the connection\n",
      "Calls: read.table -> file\n",
      "In addition: Warning message:\n",
      "In file(file, \"rt\") :\n",
      "  cannot open file '/home/cke/PseudoBulk/Results/cellcategory_subtype.csv': No such file or directory\n",
      "Execution halted\n",
      "[Parallel(n_jobs=10)]: Done 160 out of 162 | elapsed:  6.5min remaining:    4.9s\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.523650479316712\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_26_LOT_BLADEout_top20markers.pickle\n",
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.598040560881297\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_6_LOT_BLADEout_top20markers.pickle\n",
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.602287614345551\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_18_LOT_BLADEout_top20markers.pickle\n",
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.605280884106954\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_16_LOT_BLADEout_top20markers.pickle\n",
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.520261077086131\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_20_LOT_BLADEout_top20markers.pickle\n",
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 6.589730072021484\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_25_LOT_BLADEout_top20markers.pickle\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=10)]: Done 162 out of 162 | elapsed:  7.3min finished\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No feature filtering is done (fsel = 0)\n",
      "Done optimization, elapsed time (min): 7.382211204369863\n",
      "export to:  /home/cke/PseudoBulk/Results/BLADE/test_fullpipeline_1_pseudobulk_22_LOT_BLADEout_top20markers.pickle\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa2f8f79",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
